### PR TITLE
Feature/ios touch feedback

### PR DESCRIPTION
**Pull request by jorre127**

### CODE REVIEW

File: lib/src/widget/touch_feedback/touch_feedback.dart [hunk 1]
```diff
@@ -30,71 +60,70 @@ class TouchFeedBack extends StatefulWidget {
...
     return Semantics(
       label: semanticsLabel,
       button: true,
-      child: BetterInkwell(
+      child: TouchManager(
         borderRadius: borderRadius,
-        onTap: onClick,
+        onTap: onTapped,
+        tapColor: tapColor ??
+            (isAndroid
```
Comment (ikbendewilliam) [inline lib/src/widget/touch_feedback/touch_feedback.dart:111]: Also add a light and dark version for android

File: lib/src/widget/touch_feedback/touch_feedback.dart [hunk 2]
```diff
@@ -30,71 +60,70 @@ class TouchFeedBack extends StatefulWidget {
...
-  final VoidCallback? onClick;
-  final String? semanticsLabel;
-  final Color color;
-  final BorderRadius? borderRadius;
-  final double elevation;
-  final Color? shadowColor;
-  final ShapeBorder? shapeBorder;
-
-  const TouchFeedBackAndroid({
+  const TouchFeedBack.dark({
```
Comment (ikbendewilliam) [inline lib/src/widget/touch_feedback/touch_feedback.dart:67]: Don't use named constructors here, there are to many options IMO

File: lib/src/widget/touch_feedback/touch_feedback.dart [hunk 3]
```diff
@@ -30,71 +60,70 @@ class TouchFeedBack extends StatefulWidget {
...
         borderRadius: borderRadius,
-        onTap: onClick,
+        onTap: onTapped,
+        tapColor: tapColor ??
+            (isAndroid
+                ? androidTapColor
+                : isDark
+                    ? iosDarkPressColor
+                    : iosLightPressColor),
+        touchEffectBuilders: [
```
Comment (ikbendewilliam) [inline lib/src/widget/touch_feedback/touch_feedback.dart:116]: Make it possible to provide custom `touchEffectBuilders`

**Review by ikbendewilliam — CHANGES_REQUESTED** [review]
Channge major version + update documentation & changelog

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,8 @@
+# Changelog
+## [2.1.6] - 2024-02-22
+### Changed
+-  Updated touchfeedback for ios to be more inline with the specs.
```
Comment (ikbendewilliam) [inline CHANGELOG.md:4]: breaking change -> version `2.2.0`
Add breaking: `onClick` becomes `onTapped`
Removed BetterInkwell widget + explain how to use it now
`TouchFeedBack ` & `iOS` to be more inline with the native look and feel
Added option to specify custom effects

**Review by ikbendewilliam — CHANGES_REQUESTED** [review]

File: pubspec.yaml [hunk 5]
```diff
@@ -3,7 +3,7 @@ description: icapps architecture components for flutter projects. Contains commo
 homepage: https://github.com/icapps/flutter-icapps-architecture
 repository: https://github.com/icapps/flutter-icapps-architecture
 issue_tracker: https://github.com/icapps/flutter-icapps-architecture/issues
-version: 2.1.5
+version: 2.1.6
```
Comment (ikbendewilliam) [inline pubspec.yaml:6]: Also update this to 2.2.0 😅

Comment (jorre127) [reply pubspec.yaml:6]: Ah damn I was thinking don't forget to update this while updating the changelog haha

**Review by ikbendewilliam — APPROVED** [review]

**Pull request merged**